src/realm/exceptions.hpp

```diff
@@ -22,65 +22,64 @@
 #include <stdexcept>
 
 #include <realm/util/features.h>
-#include <realm/util/backtrace.hpp>
-#include <realm/util/to_string.hpp>
+#include <realm/status.hpp>
 
 namespace realm {
 
-using util::ExceptionWithBacktrace;
-
 /// Thrown by various functions to indicate that a specified table does not
 /// exist.
-class NoSuchTable : public ExceptionWithBacktrace<std::exception> {
+class NoSuchTable : public ExceptionForStatus {
```

Comment:
A lot of these exceptions are losing the stack trace which has been useful in debugging. Can we get a way to preserve the stack trace via ExceptionForStatus? Probably not all exceptions should have it though so there should be some way to opt out as well.

Reply:
Do we still find it useful? I had the impression that this was mostly to help debugging on the legacy server where we did not have core dumps. I think that in most reports on phone crashes, we have a stack trace. Let's discuss on the round table.

Reply:
After discussing this as a team, we have decided not to attempt to capture the stacktrace as part of the exception message any longer.
1) The capture is not supported on several platforms such as android.
2) We rarely get exceptions reported, normally we get assertion errors and other crashes reported which already include a stack trace.
3) There is a performance cost when attempting to capture the stack.